### ISSUE TITLE
Green screen and reboot 

### ISSUE THREAD

**Issue by EBendinelli** [seerge/g-helper#998]
I'm conscious this is not going to be an easy one to reproduce. I've been happily using GHelper since I got my G14 (2022/6900hs/6800s) but this week I encountered a green screen (on the external display I'm using) followed by a reboot while playing God of War after 30 minutes+. This happened 3 times in a row. After some investigation I disabled GHelper and re-installed Armoury Crate and have played 2h in a row without issue which makes me think the issue is from GHelper. I know there are other posts online about similar issues with AMD components (usually resolved by changing the HDMI cable or unplugging docks) but since in this case disabling GHelper was the solution I'm flagging it here. I've included the logs from when it happened (usually right before the logs indicated app start).

Windows 11 Event viewer just reports an [Event ID 41](https://learn.microsoft.com/en-us/troubleshoot/windows-client/performance/event-id-41-restart) with no helpful information on the cause of the reboot.

Happy to do some testing for you if I can help reproduce the bug 

**Clear scenario to Reproduce**
1. Install God of War on Steam
2. Play for 30 minutes, default settings
3. Green screen on the external display (computer screen is frozen) for 3 seconds followed by a reboot

**Expected behavior**
No crash

**App Logs**
```
02/08/2023 11:18:10: ------------
02/08/2023 11:18:10: App launched: ROG Zephyrus G14 GA402RK_GA402RK :0.100.0.0en-GB
02/08/2023 11:18:10: AMD Radeon RX 6800S
02/08/2023 11:18:11: CPU: 8 - AMD Ryzen 9 6900HS with Radeon Graphics         - AMD64 Family 25 Model 68 Stepping 1
02/08/2023 11:18:11: AutoSetting for Online
02/08/2023 11:18:11: Optimization service is running
02/08/2023 11:18:11: Mode = 1 : OK
02/08/2023 11:18:11: Eco flag : 0
02/08/2023 11:18:11: Mux flag : 1
02/08/2023 11:18:11: :DISPLAYCONFIG_OUTPUT_TECHNOLOGY_INTERNAL: \\.\DISPLAY1
02/08/2023 11:18:11: 23MP55:DISPLAYCONFIG_OUTPUT_TECHNOLOGY_HDMI: \\.\DISPLAY6
02/08/2023 11:18:11: Screen = 120Hz : OK
02/08/2023 11:18:11: ScreenOverdrive = 1 : OK
02/08/2023 11:18:11: BatteryLimit = 70 : OK
02/08/2023 11:18:11: BatteryLimit = 70 : OK
02/08/2023 11:18:11: USB-KB 0x19B6:5D-BD-01-55-1E-0F
02/08/2023 11:18:11: USB-KB 64|32HID-compliant vendor-defined device\\?\hid#vid_0b05&pid_19b6&mi_00&col04#8&18c99fbc&0&0003#{4d1e55b2-f16f-11cf-88cb-001111000030}:5D-B3-00-00-FF-FF-FF-EB-00-00-FF-FF-FF-00-00-00-00
02/08/2023 11:18:11: Matrix Off
02/08/2023 11:18:11: Found external screen: 23MP55:DISPLAYCONFIG_OUTPUT_TECHNOLOGY_HDMI
02/08/2023 11:18:11: Auto:5D-BA-C5-C4-00
02/08/2023 11:18:11: Changed Lid Action to 0
02/08/2023 11:18:11: Engaging Clamshell Mode
02/08/2023 11:18:11: Monitor Power On
02/08/2023 11:18:11: Display configuration changed.
02/08/2023 11:18:11: Found external screen: 23MP55:DISPLAYCONFIG_OUTPUT_TECHNOLOGY_HDMI
02/08/2023 11:18:11: Changed Lid Action to 0
02/08/2023 11:18:11: Engaging Clamshell Mode
02/08/2023 11:18:46: Monitor Power Off
02/08/2023 11:18:47: Backlight:5D-BA-C5-C4-00
02/08/2023 11:20:04: Monitor Power On
02/08/2023 11:20:04: AutoSetting for Online
02/08/2023 11:20:04: Optimization service is running
02/08/2023 11:20:04: Mode = 1 : OK
02/08/2023 11:20:04: Eco flag : 0
02/08/2023 11:20:04: Mux flag : 1
02/08/2023 11:20:04: :DISPLAYCONFIG_OUTPUT_TECHNOLOGY_INTERNAL: \\.\DISPLAY1
02/08/2023 11:20:04: 23MP55:DISPLAYCONFIG_OUTPUT_TECHNOLOGY_HDMI: \\.\DISPLAY6
02/08/2023 11:20:04: Screen = 120Hz : OK
02/08/2023 11:20:04: ScreenOverdrive = 1 : OK
02/08/2023 11:20:04: BatteryLimit = 70 : OK
02/08/2023 11:20:05: USB-KB 0x19B6:5D-BD-01-55-1E-0F
02/08/2023 11:20:05: USB-KB 64|32HID-compliant vendor-defined device\\?\hid#vid_0b05&pid_19b6&mi_00&col04#8&18c99fbc&0&0003#{4d1e55b2-f16f-11cf-88cb-001111000030}:5D-B3-00-00-FF-FF-FF-EB-00-00-FF-FF-FF-00-00-00-00
02/08/2023 11:20:05: Matrix Off
02/08/2023 11:20:05: Auto:5D-BA-C5-C4-00
02/08/2023 11:20:34: Monitor Power Off
02/08/2023 11:20:34: Backlight:5D-BA-C5-C4-00
02/08/2023 12:28:45: Monitor Power On
02/08/2023 12:28:45: AutoSetting for Online
02/08/2023 12:28:45: Optimization service is running
02/08/2023 12:28:45: Mode = 1 : OK
02/08/2023 12:28:45: Eco flag : 0
02/08/2023 12:28:45: Mux flag : 1
02/08/2023 12:28:45: :DISPLAYCONFIG_OUTPUT_TECHNOLOGY_INTERNAL: \\.\DISPLAY1
02/08/2023 12:28:45: 23MP55:DISPLAYCONFIG_OUTPUT_TECHNOLOGY_HDMI: \\.\DISPLAY6
02/08/2023 12:28:45: Screen = 120Hz : -1
02/08/2023 12:28:46: Screen = 120Hz : -1
02/08/2023 12:28:46: ScreenOverdrive = 1 : OK
02/08/2023 12:28:46: BatteryLimit = 70 : OK
02/08/2023 12:28:46: USB-KB 0x19B6:5D-BD-01-55-1E-0F
02/08/2023 12:28:46: USB-KB 64|32HID-compliant vendor-defined device\\?\hid#vid_0b05&pid_19b6&mi_00&col04#8&18c99fbc&0&0003#{4d1e55b2-f16f-11cf-88cb-001111000030}:5D-B3-00-00-FF-FF-FF-EB-00-00-FF-FF-FF-00-00-00-00
02/08/2023 12:28:47: Matrix Off
02/08/2023 12:28:47: Auto:5D-BA-C5-C4-00
02/08/2023 13:41:50: ------------
02/08/2023 13:41:50: App launched: ROG Zephyrus G14 GA402RK_GA402RK :0.100.0.0en-GB
02/08/2023 13:41:50: AMD Radeon RX 6800S
02/08/2023 13:41:52: CPU: 8 - AMD Ryzen 9 6900HS with Radeon Graphics         - AMD64 Family 25 Model 68 Stepping 1
02/08/2023 13:41:52: AutoSetting for Online
02/08/2023 13:41:52: Optimization service is running
02/08/2023 13:41:52: Mode = 1 : OK
02/08/2023 13:41:52: Eco flag : 0
02/08/2023 13:41:52: Mux flag : 1
02/08/2023 13:41:52: :DISPLAYCONFIG_OUTPUT_TECHNOLOGY_INTERNAL: \\.\DISPLAY1
02/08/2023 13:41:52: 23MP55:DISPLAYCONFIG_OUTPUT_TECHNOLOGY_HDMI: \\.\DISPLAY6
02/08/2023 13:41:52: Screen = 120Hz : OK
02/08/2023 13:41:52: ScreenOverdrive = 1 : OK
02/08/2023 13:41:52: BatteryLimit = 70 : OK
02/08/2023 13:41:52: BatteryLimit = 70 : OK
02/08/2023 13:41:52: USB-KB 0x19B6:5D-BD-01-55-1E-0F
02/08/2023 13:41:52: USB-KB 64|32HID-compliant vendor-defined device\\?\hid#vid_0b05&pid_19b6&mi_00&col04#8&18c99fbc&0&0003#{4d1e55b2-f16f-11cf-88cb-001111000030}:5D-B3-00-00-FF-FF-FF-EB-00-00-FF-FF-FF-00-00-00-00
02/08/2023 13:41:52: Matrix Off
02/08/2023 13:41:52: Found external screen: 23MP55:DISPLAYCONFIG_OUTPUT_TECHNOLOGY_HDMI
02/08/2023 13:41:52: Auto:5D-BA-C5-C4-00
02/08/2023 13:41:52: Changed Lid Action to 0
02/08/2023 13:41:52: Engaging Clamshell Mode
02/08/2023 13:41:52: Monitor Power On
02/08/2023 13:41:52: Display configuration changed.
02/08/2023 13:41:52: Found external screen: 23MP55:DISPLAYCONFIG_OUTPUT_TECHNOLOGY_HDMI
02/08/2023 13:41:52: Changed Lid Action to 0
02/08/2023 13:41:52: Engaging Clamshell Mode
02/08/2023 13:41:58: Design Capacity: 76000mWh, Full Charge Capacity: 74136mWh, Health: 0.9754736842105263157894736842%
02/08/2023 13:42:03: https://github.com/seerge/g-helper/releases/download/v0.103/GHelper.zip
02/08/2023 13:42:03: C:\Users\BENDY\Documents\apps\GHelper\GHelper.zip
02/08/2023 13:42:03: C:\Users\BENDY\Documents\apps\GHelper\GHelper.exe
02/08/2023 13:42:09: ------------
02/08/2023 13:42:09: App launched: ROG Zephyrus G14 GA402RK_GA402RK :0.103.0.0en-GB
02/08/2023 13:42:09: AMD Radeon RX 6800S
02/08/2023 13:42:10: CPU: 8 - AMD Ryzen 9 6900HS with Radeon Graphics         - AMD64 Family 25 Model 68 Stepping 1
02/08/2023 13:42:10: AutoSetting for Online
02/08/2023 13:42:10: Optimization service is running
02/08/2023 13:42:10: Mode = 1 : OK
02/08/2023 13:42:10: Eco flag : 0
02/08/2023 13:42:10: Mux flag : 1
02/08/2023 13:42:10: :DISPLAYCONFIG_OUTPUT_TECHNOLOGY_INTERNAL: \\.\DISPLAY1
02/08/2023 13:42:10: 23MP55:DISPLAYCONFIG_OUTPUT_TECHNOLOGY_HDMI: \\.\DISPLAY6
02/08/2023 13:42:10: Screen = 120Hz : OK
02/08/2023 13:42:10: ScreenOverdrive = 1 : OK
02/08/2023 13:42:10: BatteryLimit = 70 : OK
02/08/2023 13:42:10: BatteryLimit = 70 : OK
02/08/2023 13:42:10: Matrix Off
02/08/2023 13:42:10: Monitor Power On
02/08/2023 13:42:10: USB-KB 0x19B6:5D-BD-01-55-1E-FF-FF-FF
02/08/2023 13:42:10: USB-KB 64|32HID-compliant vendor-defined device\\?\hid#vid_0b05&pid_19b6&mi_00&col04#8&18c99fbc&0&0003#{4d1e55b2-f16f-11cf-88cb-001111000030}:5D-B3-00-00-FF-FF-FF-EB-00-00-FF-FF-FF-00-00-00-00
02/08/2023 13:42:34: Design Capacity: 76000mWh, Full Charge Capacity: 74136mWh, Health: 0.9754736842105263157894736842%
02/08/2023 13:42:34: Latest version 0.103.0.0
02/08/2023 14:11:01: ------------
02/08/2023 14:11:01: App launched: ROG Zephyrus G14 GA402RK_GA402RK :0.103.0.0en-GB
02/08/2023 14:11:01: AMD Radeon RX 6800S
02/08/2023 14:11:03: CPU: 8 - AMD Ryzen 9 6900HS with Radeon Graphics         - AMD64 Family 25 Model 68 Stepping 1
02/08/2023 14:11:03: AutoSetting for Online
02/08/2023 14:11:03: Optimization service is running
02/08/2023 14:11:03: Mode = 1 : OK
02/08/2023 14:11:03: Eco flag : 0
02/08/2023 14:11:03: Mux flag : 1
02/08/2023 14:11:03: :DISPLAYCONFIG_OUTPUT_TECHNOLOGY_INTERNAL: \\.\DISPLAY1
02/08/2023 14:11:03: 23MP55:DISPLAYCONFIG_OUTPUT_TECHNOLOGY_HDMI: \\.\DISPLAY6
02/08/2023 14:11:03: Screen = 120Hz : OK
02/08/2023 14:11:03: ScreenOverdrive = 1 : OK
02/08/2023 14:11:03: BatteryLimit = 70 : OK
02/08/2023 14:11:03: BatteryLimit = 70 : OK
02/08/2023 14:11:03: Matrix Off
02/08/2023 14:11:03: Monitor Power On
02/08/2023 14:11:03: Display configuration changed.
02/08/2023 14:11:03: USB-KB 0x19B6:5D-BD-01-55-1E-FF-FF-FF
02/08/2023 14:11:03: Auto:5D-BA-C5-C4-00
02/08/2023 14:11:03: USB-KB 64|32HID-compliant vendor-defined device\\?\hid#vid_0b05&pid_19b6&mi_00&col04#8&18c99fbc&0&0003#{4d1e55b2-f16f-11cf-88cb-001111000030}:5D-B3-00-00-FF-FF-FF-EB-00-00-FF-FF-FF-00-00-00-00
02/08/2023 14:11:04: Power Mode:StatusChange
02/08/2023 14:12:53: Latest version 0.103.0.0
02/08/2023 14:12:54: Design Capacity: 76000mWh, Full Charge Capacity: 74136mWh, Health: 0.9754736842105263157894736842%
02/08/2023 14:15:26: Mode = 0 : OK
02/08/2023 14:15:26: Power mode: 00000000-0000-0000-0000-000000000000
02/08/2023 14:16:55: Mode = 1 : OK
02/08/2023 14:16:55: Power mode: ded574b5-45a0-4f42-8737-46345c09c238
02/08/2023 14:17:14: Varibright: 1,2
02/08/2023 14:18:05: Boost 3
02/08/2023 14:35:15: Design Capacity: 76000mWh, Full Charge Capacity: 74136mWh, Health: 0.9754736842105263157894736842%
02/08/2023 14:35:22: Boost 4
02/08/2023 14:42:33: HID Device Event: Checking for new ASUS Mice
02/08/2023 15:04:05: Monitor Dimmed
02/08/2023 15:04:20: Monitor Power Off
02/08/2023 15:04:20: Backlight:5D-BA-C5-C4-00
02/08/2023 16:06:06: Monitor Power On
02/08/2023 16:06:06: AutoSetting for Online
02/08/2023 16:06:06: Optimization service is running
02/08/2023 16:06:06: Mode = 1 : OK
02/08/2023 16:06:06: Eco flag : 0
02/08/2023 16:06:06: Mux flag : 1
02/08/2023 16:06:06: :DISPLAYCONFIG_OUTPUT_TECHNOLOGY_INTERNAL: \\.\DISPLAY1
02/08/2023 16:06:06: 23MP55:DISPLAYCONFIG_OUTPUT_TECHNOLOGY_HDMI: \\.\DISPLAY6
02/08/2023 16:06:06: Screen = 120Hz : -1
02/08/2023 16:06:06: Screen = 120Hz : -1
02/08/2023 16:06:06: ScreenOverdrive = 1 : OK
02/08/2023 16:06:06: BatteryLimit = 70 : OK
02/08/2023 16:06:06: Matrix Off
02/08/2023 16:06:06: Auto:5D-BA-C5-C4-00
02/08/2023 16:06:06: USB-KB 0x19B6:5D-BD-01-55-1E-FF-FF-FF
02/08/2023 16:06:06: USB-KB 64|32HID-compliant vendor-defined device\\?\hid#vid_0b05&pid_19b6&mi_00&col04#8&18c99fbc&0&0003#{4d1e55b2-f16f-11cf-88cb-001111000030}:5D-B3-00-00-FF-FF-FF-EB-00-00-FF-FF-FF-00-00-00-00
02/08/2023 16:06:18: Session:SessionUnlock
02/08/2023 16:06:18: :DISPLAYCONFIG_OUTPUT_TECHNOLOGY_INTERNAL: \\.\DISPLAY1
02/08/2023 16:06:18: 23MP55:DISPLAYCONFIG_OUTPUT_TECHNOLOGY_HDMI: \\.\DISPLAY6
02/08/2023 16:06:18: Screen = 120Hz : OK
02/08/2023 16:06:18: ScreenOverdrive = 1 : OK
02/08/2023 16:09:56: HID Device Event: Checking for new ASUS Mice
```

**Desktop:**
- OS: Windows 11

**Asus software**
None

**GHelper settings**
- Turbo
- Optimized

**Comment by seerge** [comment]
@EBendinelli hello, I honestly don't know why your laptop rebooted, i would rather attribute it to some hiccup with drivers or system in general (mb try to do a fresh install from AMD using AMD cleanup utility?) You have same device as I personally have, and I also can play quite intensive games for a long time w/o any issues :) 

G-Helper is not a driver, and when running in background it literally does nothing (it only sets one of 3 predefined modes in bios, and can optionally add power limits / fan curve). 

Did you undervolt or set any other settings in the app? (log covers only for August 2, so i don't see it there)

**Comment by seerge** [comment]
P.S. I have noticed that you have at least Asus Optimization Service running : 
```
02/08/2023 16:06:06: Optimization service is running
```

I don't think this is a reason, but for the clean experiment I would stop all of them from Extra -> Stop (as they are not needed for g-helper, and needed only for MyASUS to work) + you can start them back from same button :)

**Comment by seerge** [comment]
@EBendinelli hello, any updates? :)

**Comment by EBendinelli** [comment]
Sorry I forgot to follow up! The problem just fixed itself so not much to add. For reference I had reinstalled the GPU drivers manually (not using the cleanup utility) while still using G-helper and that didn't help. I disabled G-helper, re-installed Armoury Crate and didn't get any issue. A few days later I removed Armoury Crate, re-enabled GHelper and killed all Asus services and it's still functioning perfectly. I've a played a few hours since then without any problem so it's all sorted!

Taking this opportunity to thank you for you work, it's quite an amazing little piece of software you've written here, really grateful for your work!
